### PR TITLE
Automatically Create Catagory "Index" Pages

### LINKED ISSUE #54: [Prototype] Add auto-generated directory index pages
## Summary
Automatically generate directory index pages for directories that do not already have an authored index.mdx (landing page) that are super basic, roughly similar to existing index pages.

## Acceptance criteria
- [ ] In gatsby-node.js, map relationship for mdx files to directories, directories to other directories
- [ ] Create template to handle the data you pass to it and render content (likely need to use a different template than the basic one created to display other .mdx files)

## Helpful links
[Docs IA](https://whimsical.com/8tBfw8Squs35bvTo5acsC6)
[Example from my prototype](https://github.com/roadlittledawn/gatsby-netlify-cms-example/blob/master/gatsby-node.js)

**Directory index page examples**
https://docs.newrelic.com/docs/synthetics?toc=true
https://docs.newrelic.com/docs/agents/php-agent/getting-started

### PR DESCRIPTION
**Pull request by zstix**
## Description
When we run a migration, we wanted to created an `index.mdx` file in every directory that served as a "landing page" for that category. The pages have links to all the pages in that category and any sub-categories. If an index page already exists, the script will skip creating a file.

There is a lot of cleanup that could happen with the generated content (see screenshot), but the basic functionality is in place.

This PR also adds a little logging utility. Feel free to improve or remove it.

## Related Issue(s)
* Closes #54 

## Screenshot(s)
Screen Shot 2020-09-14 at 4 27 45 PM

### CODE REVIEW

File: scripts/create-index-pages.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+---
+
+`;
+
+const getPageContent = (dir, files) => {
+  return files
+    .map((file) => {
+      const path = `${dir}/${file.name}`.replace(BASE_DIR, '');
+      // TODO: get the label from frontmatter
+      const label = getTitle(file.name);
```
Comment (zstix) [inline scripts/create-index-pages.js:29]: At the moment, this takes the last part of the filepath and turns it into a label. We probably want to grab the `title` from the frontmatter for that file and use that instead. If someone wants to take a crack at this, that would be rad.

**Comment by roadlittledawn** [comment]
fyi. pushed quick change to remove .mdx from link paths in index.mdx files so they're resolvable when running site locally.

File: scripts/create-index-pages.js (hunk 1 above)
Comment (caylahamann) [reply scripts/create-index-pages.js:31]: I'm working on this!

File: scripts/create-index-pages.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+`;
+
+const getPageContent = (dir, files) => {
+  return files
+    .map((file) => {
+      const path = `${dir}/${file.name}`
+        .replace(BASE_DIR, '')
+        .replace('.mdx', '');
+      const fileContent = fsSync.readFileSync(`${dir}/${file.name}`, 'utf8');
```
Comment (caylahamann) [inline scripts/create-index-pages.js:31]: @zstix @roadlittledawn attempted to have it read from the actual title! appears to be working

Comment (roadlittledawn) [reply scripts/create-index-pages.js:31]: yep, looks good to me! good stuff. i like that we maintain a fallback to the file name, too.

**Review by roadlittledawn — APPROVED** [review]
:shipit:

**Pull request merged**